dart-mode.el

```diff
@@ -839,6 +862,40 @@ true for positions before the start of the statement, but on its line."
          ((?{) (dart-in-block-p (c-guess-basic-syntax))))))))
 
 
+(defun dart--process-nav-info (response offset)
+  "Report the possible completions and jump to the location of the definition.
+
+  Opens a new file in a new buffer if necessary."
+  (dart-info (format "Reporting navigation : %s" response))
+  (-when-let* ((regions (cdr (assq 'regions (assq 'result response))))
+	       (target-offset
+		(loop for region across regions
+		      if (equal offset (cdr (assoc 'offset region)))
```

Comment:
Nit: it's confusing that you're using `assoc` in some places and `assq` in others. I'd just use `assoc` everywhere, since that's what the rest of the library does.